### ISSUE TITLE
SegFault 11 when empty H1 H2 H3 H4 etc element is used in insert_htmlbox

### ISSUE THREAD

**Issue by cs-shadowbq** [pymupdf/PyMuPDF#3559]
### Description of the bug

SegFault 11 when empty HTML elements H1 H2 H3 H4 are used in insert_htmlbox.

### How to reproduce the bug

Assume
```python
import pymupdf
doc = pymupdf.Document()
page = doc.new_page()
```

This Fails
```python
>>> text_insert="""<body><h3></h3></body>"""
>>> page.insert_htmlbox(fullrect, text_insert)
Segmentation fault: 11
```

This passes
```python
>>> text_insert="""<body><h3>&nbsp;</h3></body>"""
>>> page.insert_htmlbox(fullrect, text_insert)
(701.0720024108887, 1.0)
```

Other html elements do NOT have this failure such as (div, span,b,center..)

### PyMuPDF version

1.24.3

### Operating system

MacOS

### Python version

3.8

**Comment by JorjMcKie** [comment]
Confirmed to also happen on Windows.

**Comment by cs-shadowbq** [comment]
upstream 👀  - https://github.com/ArtifexSoftware/mupdf/commit/b084345acbbef5de2fafd310278e8f73d9269c2b

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.24.6.
